### PR TITLE
Use LocationAwareConfigRepository from pimcore/pimcore:^10.2

### PR DESCRIPTION
**Pull request by mcop1**
Part of #421

### CODE REVIEW

File: src/Configuration.php [hunk 1]
```diff
@@ -95,6 +156,9 @@ public function setConfiguration($configuration): void
         if (is_array($configuration)) {
             $configuration = json_decode(json_encode($configuration), true);
         }
+        if (empty($this->getName()) == true) {
```
Comment (NiklasBr) [inline src/Configuration.php]:
```suggestion
        if (empty($this->getName())) {
```
It's the same thing :)

File: src/Configuration/Dao.php [hunk 2]
```diff
@@ -39,88 +40,146 @@ class Dao extends AbstractDao
...
     public static function getConfigModificationDate()
     {
-        $config = Config::locateConfigFile(self::CONFIG_FILE);
-
-        if (!file_exists($config)) {
-            return false;
-        }
-
-        return filemtime($config);
+        return strtotime('1987-10-09 13:58:00');
```
Comment (NiklasBr) [inline src/Configuration/Dao.php]: This seems odd

File: src/Configuration.php (hunk 1 above)
Comment (mcop1) [reply src/Configuration.php]: Thanks for the suggestion and true- it is a bit of habit I developed over the years :)

**Comment by fashxp** [comment]
please also add note to the docs ... similar to other bundles that are using the location aware config

File: src/Configuration/Dao.php [hunk 3]
```diff
@@ -39,88 +40,146 @@ class Dao extends AbstractDao
...
     public static function getConfigModificationDate()
     {
-        $config = Config::locateConfigFile(self::CONFIG_FILE);
-
-        if (!file_exists($config)) {
-            return false;
-        }
-
-        return filemtime($config);
+        return strtotime('1970-01-01 00:00:00');
```
Comment (NiklasBr) [inline src/Configuration/Dao.php]:
```suggestion
        return 0;
```

File: doc/20_Deployment.md [hunk 4]
```diff
@@ -21,4 +21,48 @@ to do that for all definitions, or
...
+
+### Configuration Storage
+
+The configuration user interface utilizes the `LocationAwareConfigRepository` for storing the configuration. With
+the environment variable `PIMCORE_WRITE_TARGET_DATA_HUB` the storage location can be configured, possible values are
+- `symfony-config` - write configs as Symfony Config as YAML files to `/var/config/data-hub/<name>.yaml`
+- `settings-store` - write configs to the SettingsStore
+- `disabled` - do not allow to edit/write configs at all
+
+Details also see [Pimcore Docs](https://pimcore.com/docs/pimcore/10.2/Development_Documentation/Deployment/Configuration_Environments.html#page_Configuration-Storage-Locations-Fallbacks).
```
Comment (dvesh3) [inline doc/20_Deployment.md]:
```suggestion
Details also see [Pimcore Docs](https://pimcore.com/docs/pimcore/current/Development_Documentation/Deployment/Configuration_Environments.html#page_Configuration-Storage-Locations-Fallbacks).
```

File: src/Configuration/DatahubConfigLocator.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\DataHubBundle\Configuration;
+
+use Symfony\Component\Finder\Finder;
+
+/**
+ * Locates configs from path config/pimcore/headless-definitions.
```
Comment (dvesh3) [inline src/Configuration/DatahubConfigLocator.php]:
```suggestion
 * Locates data hub configs
```

File: src/Resources/public/js/configuration/graphql/configItem.js [hunk 6]
```diff
@@ -15,11 +15,16 @@ pimcore.registerNS("pimcore.plugin.datahub.configuration.graphql.configItem");
...
     saveUrl: "/admin/pimcoredatahub/config/save",
+    disableForm: false,
 
     initialize: function (data, parent) {
         this.parent = parent;
         this.data = data.configuration;
         this.modificationDate = data.modificationDate;
+        if(data['configuration']['general']['writeable'] != null &&
+            data['configuration']['general']['writeable'] != undefined) {
+            this.disableForm = !data['configuration']['general']['writeable'];
```
Comment (dvesh3) [inline src/Resources/public/js/configuration/graphql/configItem.js]: writeable check should be negated here..right?

```suggestion
            this.disableForm = data['configuration']['general']['writeable'];
```

Comment (dvesh3) [reply src/Resources/public/js/configuration/graphql/configItem.js]: Nevermind...it is used in the disabled property so makes sense!

Comment (dvesh3) [reply src/Resources/public/js/configuration/graphql/configItem.js]: Ok...save button should consider this check at line 111

Comment (mcop1) [reply src/Resources/public/js/configuration/graphql/configItem.js]: When we disable the save button, the user would never get feedback on why he can´t save the configuration. Should we really do this? Wdyt?

Comment (dvesh3) [reply src/Resources/public/js/configuration/graphql/configItem.js]: We've done this for thumbnails configuration:
image

And it is better than making every field read only and then consistent with other features.

**Review by dvesh3 — APPROVED** [review]

**Pull request merged**